### PR TITLE
Documentation: add group inheritance chapter

### PR DESCRIPTION
**Pull request by zbbfufu**
Hibernate-validator documentation do not mention the possibility to declare constraints group inheriting from other groups or from Default group, as describe in [bean-validation 1.1 spec](http://beanvalidation.org/1.1/spec/#constraintdeclarationvalidationprocess-groupsequence-groupinheritance).
This PR proposes a chapter about group inheritance with examples based on Car class.

### CODE REVIEW

File: documentation/src/main/asciidoc/ch05.asciidoc [hunk 1]
```diff
@@ -106,6 +106,49 @@ due to the fact that the driver has not yet passed the driving test. Only after
...
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter05/GroupTest.java[tags=supercar]
+----
+====
+
+On the first call to `validate()`, we do not specify a group. There is one validation error because a car must have at least one seat.
+It's a constraint from the Default group.
+
+On the second call, we specify only the group RaceCarChecks. There are two validation errors: one about the seat from Default group,
```
Comment (gsmet) [inline documentation/src/main/asciidoc/ch05.asciidoc]: from **the** `Default` group

File: documentation/src/main/asciidoc/ch05.asciidoc [hunk 2]
```diff
@@ -106,6 +106,49 @@ due to the fact that the driver has not yet passed the driving test. Only after
...
+another one about safety belt from RaceCarChecks group.
```
Comment (gsmet) [inline documentation/src/main/asciidoc/ch05.asciidoc]: from **the** `RaceCarChecks` group

File: documentation/src/main/asciidoc/ch05.asciidoc [hunk 3]
```diff
@@ -106,6 +106,49 @@ due to the fact that the driver has not yet passed the driving test. Only after
...
+
+
```
Comment (gsmet) [inline documentation/src/main/asciidoc/ch05.asciidoc]: Only one empty line is enough.

File: documentation/src/main/asciidoc/ch05.asciidoc [hunk 4]
```diff
@@ -106,6 +106,49 @@ due to the fact that the driver has not yet passed the driving test. Only after
...
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/org/hibernate/validator/referenceguide/chapter05/groupinheritance/RaceCarChecks.java[]
+----
+====
+
+Now we will check if a SuperCar with one seat and no security belt is a valid car and a valid race-car.
+
+
```
Comment (gsmet) [inline documentation/src/main/asciidoc/ch05.asciidoc]: Only one empty line is enough.

**Comment by zbbfufu** [comment]
Created jiras:
[HV-1132](https://hibernate.atlassian.net/browse/HV-1132) force language
[HV-1133](https://hibernate.atlassian.net/browse/HV-1133) document group inheritance

**Comment by gsmet** [comment]
@zbbfufu do you plan to fix the issues I mentioned or should I do it? Thanks!

**Comment by zbbfufu** [comment]
@gsmet I'll fix it. I would like to refactor my commits to have one for each Jira.

**Comment by zbbfufu** [comment]
I pushed a rewrite of commits, one for each issue. It takes you comments into account.

**Comment by gsmet** [comment]
@zbbfufu I added one more commit for some minor wording improvements and merged!

Thanks!

**Comment by gsmet** [comment]
@zbbfufu btw, could you sign the CLA for the Hibernate Validator project? It's a simple form to fill at https://cla.jboss.org . Thanks!

**Comment by zbbfufu** [comment]
@gsmet I signed the license agreement. I created my JBoss Developer account
as jfurgerot.

**Pull request closed without merging**